packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -122,6 +123,10 @@ export interface UnifiedDataTableProps {
    * Check available utils in `utils/get_column_types.ts`
    */
   columnTypes?: DataTableColumnTypes;
+  /**
+   * An optional settings for customising the column
+   */
+  customiseColumn?: CustomColumn;
```

Comment:
Something like `customGridColumnsConfiguration` would be more readable imho. And move it next to `externalCustomRenderers`.